activerecord/test/cases/calculations_test.rb

```diff
@@ -986,11 +1006,27 @@ def test_ids_with_includes
     assert_equal Company.all.map(&:id).sort, Company.all.includes(:contracts).ids.sort
   end
 
-  def test_ids_with_scope_and_includes
+  def test_ids_with_includes_and_scope
     scoped_ids = [1, 2]
     company = Company.where(id: scoped_ids).first
     5.times { company.contracts.create! }
-    assert_equal Company.where(id: scoped_ids).map(&:id).sort, Company.where(id: scoped_ids).includes(:contracts).ids.sort
+    assert_equal Company.where(id: scoped_ids).map(&:id).sort, Company.includes(:contracts).where(id: scoped_ids).ids.sort
+  end
+
+  def test_ids_with_includes_and_table_scope
+    company = Company.first
+    company.contracts.create!
+    assert_equal [company.id], Company.includes(:contracts).where("contracts.id" => company.contracts.first).ids
```

Comment:
This errors without the distinct join added in `#ids`:
<img width="1115" alt="Screenshot 2022-12-10 at 4 28 53 pm" src="https://user-images.githubusercontent.com/54629302/206851690-5b48873f-e062-48ae-b400-b19b14f96282.png">